src/loggers.jl

```diff
@@ -430,8 +430,18 @@ let
                     end
 
                     function $level(msg::Function, logger::Logger)
-                        log(msg, logger, $key)
-                        throw(ErrorException(msg()))
+                        # NOTE: We can't lazily evaluate do-blocks on errors or higher
+                        # because we need to determine whether the function returns
+                        # a string or an error type. We provide this syntax for readability
+                        # rather than performance.
+                        m = msg()
+                        if isa(m, String)
```

Comment:
```suggestion
                        if m isa AbstractString
```

Reply:
Not a fan of the infix syntax

Reply:
That's fine. Should still use `AbstractString`
```suggestion
                        if isa(m, AbstractString)
```